File: src/main/java/com/zack/word/WordCount.java
package com.zack.word;

import org.apache.flink.api.common.io.InputFormat;
import org.apache.flink.api.java.ExecutionEnvironment;
import org.apache.flink.api.java.operators.DataSource;

/**
 * com.zack
 *
 * @author zhangc
 * @version 1.0
 * @create 2020/11/2 17:23
 */
public class WordCount {
    public static void main(String[] args) {
        ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
        String filePath = "D:\\IdeaProjects\\flink-demo\\src\\main\\resources\\hello.txt";
        DataSource<String> wordCountDataSet = env.readTextFile(filePath);
        wordCountDataSet.getSplitDataProperties().splitsGroupedBy();

    }
}

File: src/main/scala/WordCount.scala
import org.apache.flink.api.scala._

/**
 *
 *
 * @author zhangc
 * @version 1.0
 * @create 2020/10/29 17:02 
 */
object WordCount {
  def main(args: Array[String]): Unit = {
    val env = ExecutionEnvironment.getExecutionEnvironment

    val filePath = "D:\\IdeaProjects\\flink-demo\\src\\main\\resources\\hello.txt"
    val inputDataSet = env.readTextFile(filePath)
    val wordCountDataSet = inputDataSet.flatMap(_.split(" "))
        .map( (_, 1) )
        .groupBy(0)
        .sum(1)
    wordCountDataSet.print()
  }
}
